### PR TITLE
fix(partition): elimination of zero values

### LINKED ISSUE #642: Treemap in infinite loop when `valueAccessor` returns zero
This is a bug in the treemap only, not in the sunburst chart. To reproduce:

1. Modify one of the 2-layer treemap examples to sometimes return 0 in the valueAccessor function, such as `valueAccessor={(d: Datum) => Math.random() < 0.7 ? d.exportVal as number : 0}`
2. Refresh the storybook
3. You will see that the chart throws an infinite loop:

```
RangeError: Maximum call stack size exceeded
    at Function.get [Symbol.species] ()
    at Array.concat ()
    at push../src/chart_types/partition_chart/layout/utils/treemap.ts.__spread 
```

When I change the exact same chart to `partitionLayout: PartitionLayout.sunburst`, the chart renders.

### PR DESCRIPTION
**Pull request by monfera**
## Summary

Instead of eliminating negative values, zero values are also eliminated.
Fixes #642 
Addresses last item of the minimum expectations checklist in #612

### Checklist

Delete any items that are not applicable to this PR.

- [ ] ~~Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)~~
- [X] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [X] Proper documentation or storybook story was added for features that require explanation or tutorials
- [X] Unit tests were updated or added to match the most common scenarios

### CODE REVIEW

**Comment by monfera** [comment]
Jenkins retest this

**Comment by monfera** [comment]
Jenkins retest this please

File: src/chart_types/partition_chart/layout/viewmodel/hierarchy_of_arrays.test.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License. */
+
+import { getHierarchyOfArrays } from './hierarchy_of_arrays';
+
+const rawFacts = [
+  { sitc1: '7', exportVal: 0 },
+  { sitc1: '3', exportVal: 1929578418424 },
```
Comment (markov00) [inline src/chart_types/partition_chart/layout/viewmodel/hierarchy_of_arrays.test.ts:23]: nit: you could also include a test against negative values

Comment (monfera) [reply src/chart_types/partition_chart/layout/viewmodel/hierarchy_of_arrays.test.ts]: added it with https://github.com/elastic/elastic-charts/pull/658/commits/019323a228151212a7efc1ff62f93c24b0dab71d
